opm/autodiff/BlackoilModelEbos.hpp

```diff
@@ -626,8 +620,8 @@ namespace Opm {
                 p = std::max(p, 0.0);
 
                 // Saturation updates.
-                const double dsw = active_[Water] ? dx[cell_idx][Indices::waterSaturationIdx] : 0.0;
-                const double dxvar = active_[Gas] ? dx[cell_idx][Indices::compositionSwitchIdx] : 0.0;
+                const double dsw = FluidSystem::phaseIsActive(FluidSystem::waterPhaseIdx) ? dx[cell_idx][Indices::waterSaturationIdx] : 0.0;
```

Comment:
What is the difference between using `FluidSystem::waterPhaseIdx` and `Indices::waterSaturationIdx`?

Reply:
waterPhaseIdx is the phase index used to get the water properties from the fluidsystem etc. 
waterSaturationIdx is the index of the water equation. 
They do not necessary match.

Reply:
I think that we should (not this PR though) give all equations indices named something with `...EqIdx` similar to `conti0EqIdx`. And the canonical indices for saturations should be separate (but equality could be enforces in blackoilindices.hh). Agree @totto82 and @andlaus?

Reply:
what exactly do you mean? For performance reasons, the number of phases and components in fluid states must probably continue to be compile-time determined, otherwise I agree if the canonical indices stay the ones which do not get any special designation. (if the reverse path is chosen, what can be done about models that do not know the concept of "inactive" phases.)